### PR TITLE
Allow fixtures in suite.xml to be updated on app update

### PR DESCRIPTION
**Pull request by phillipm**
Fixtures in the suite.xml file of an app don't get updated when the app is updated. This causes issues for the scheduler feature used by ICDS.
 
cross-request: https://github.com/dimagi/commcare-android/pull/1391
Fix for http://manage.dimagi.com/default.asp?178966

### CODE REVIEW

File: backend/src/org/commcare/xml/SuiteParser.java [hunk 1]
```diff
@@ -118,9 +130,10 @@ public Suite parse() throws InvalidStructureException, IOException,
                         Menu m = new MenuParser(parser).parse();
                         menus.addElement(m);
                     } else if (parser.getName().toLowerCase().equals("fixture")) {
-                        //this one automatically commits the fixture to the global memory
-                        if (!inValidationMode()) {
-                            new FixtureXmlParser(parser, false, fixtureStorage).parse();
+                        if (!isValidationPass) {
+                            // commit fixture to the memory, overwriting existing
+                            // fixture only during install, not init or validation
+                            new FixtureXmlParser(parser, !skipResources, fixtureStorage).parse();
```
Comment (phillipm) [inline backend/src/org/commcare/xml/SuiteParser.java]: The fix is to pass `!skipResources` in as the `overwrite` argument to the fixture parser. `!skipResources` is `true` only when suite file is installed. It is `false` on suite initialization and validation passes.

**Comment by proteusvacuum** [comment]
:balloon:

File: backend/src/org/commcare/resources/ResourceManager.java [hunk 2]
```diff
@@ -232,7 +232,7 @@ public void upgrade()
                 //everything here? Locale files are registered elsewhere, and we
                 //can't guarantee we're the only thing in there, so we can't
                 //straight up clear it...
-                platform.initialize(masterTable);
+                platform.initialize(masterTable, true);
```
Comment (ctsims) [inline backend/src/org/commcare/resources/ResourceManager.java]: This step isn't really guaranteed to happen. IE: If the upgrade is successful and the app crashes in platform.clearAppState(), we won't process the new fixture.

Comment (phillipm) [inline backend/src/org/commcare/resources/ResourceManager.java]: Ah, good call. I added a comment in the code warning for this and we can follow up in another PR to fix it.

**Pull request merged**